docs/proposals/distributed_architecture/docs/01-Raft_Consensus_Integration.md

```diff
@@ -0,0 +1,43 @@
+## Raft Consensus Integration
+
+### Gap Definition and Improvement Objectives
+
+Currently, Flintlock operates with isolated state per host, lacking a unified, cluster-wide coordination mechanism. Integrating Raft consensus addresses this gap by ensuring reliable leader election, consistent log replication, and synchronized state management across the cluster.
```

Comment:
Raft could be used on the layer that does the distributed scheduling, but not in flintlock itself.

One option would be to just use etcd for storege and leader election. Might be easier than using a Raft package (like  https://github.com/etcd-io/raft) directly.